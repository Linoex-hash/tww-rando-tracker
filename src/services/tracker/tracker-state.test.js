import TrackerState from './tracker-state';
import Locations from '../logic/locations';

const LOCATIONS_LIST = [
  {
    generalLocation: 'Outset Island',
    detailedLocation: 'Savage Labyrinth - Floor 30'
  },
  {
    generalLocation: 'Dragon Roost Cavern',
    detailedLocation: "Bird's Nest"
  }
];

describe('TrackerState', () => {
  describe('initialize', () => {
    beforeEach(() => {
      jest.spyOn(Locations, 'allLocations').mockReturnValue(LOCATIONS_LIST);
    });

    test('initializes the entrances as an empty object', () => {
      TrackerState.initialize();

      expect(TrackerState.entrances).toEqual({});
    });

    test('initializes the items', () => {
      TrackerState.initialize();

      expect(TrackerState.items).toMatchSnapshot();
    });

    test('initializes the checked locations', () => {
      TrackerState.initialize();

      expect(TrackerState.locationsChecked).toMatchSnapshot();
    });
  });
});
