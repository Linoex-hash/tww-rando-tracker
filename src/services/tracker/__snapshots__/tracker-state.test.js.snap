// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackerState default initializes the checked locations 1`] = `
Object {
  "Dragon Roost Cavern": Object {
    "Bird's Nest": false,
  },
  "Outset Island": Object {
    "Savage Labyrinth - Floor 30": false,
  },
}
`;

exports[`TrackerState default initializes the items 1`] = `
Object {
  "Bait Bag": 0,
  "Ballad of Gales": 0,
  "Boat's Sail": 0,
  "Bombs": 0,
  "Boomerang": 0,
  "Cabana Deed": 0,
  "Command Melody": 0,
  "DRC Big Key": 0,
  "DRC Small Key": 0,
  "Deku Leaf": 0,
  "Delivery Bag": 0,
  "Din's Pearl": 0,
  "ET Big Key": 0,
  "ET Small Key": 0,
  "Earth God's Lyric": 0,
  "Empty Bottle": 0,
  "FW Big Key": 0,
  "FW Small Key": 0,
  "Farore's Pearl": 0,
  "Ghost Ship Chart": 0,
  "Grappling Hook": 0,
  "Hero's Charm": 0,
  "Hero's Shield": 0,
  "Hookshot": 0,
  "Hurricane Spin": 0,
  "Iron Boots": 0,
  "Maggie's Letter": 0,
  "Magic Armor": 0,
  "Magic Meter Upgrade": 0,
  "Mirror Shield": 0,
  "Moblin's Letter": 0,
  "Nayru's Pearl": 0,
  "Note to Mom": 0,
  "Power Bracelets": 0,
  "Progressive Bomb Bag": 0,
  "Progressive Bow": 0,
  "Progressive Picto Box": 0,
  "Progressive Quiver": 0,
  "Progressive Sword": 0,
  "Progressive Wallet": 0,
  "Skull Hammer": 0,
  "Song of Passing": 0,
  "Spoils Bag": 0,
  "Telescope": 0,
  "Tingle Statue": 0,
  "Tingle Tuner": 0,
  "TotG Big Key": 0,
  "TotG Small Key": 0,
  "Triforce Shard": 0,
  "WT Big Key": 0,
  "WT Small Key": 0,
  "Wind God's Aria": 0,
  "Wind Waker": 0,
  "Wind's Requiem": 0,
}
`;
